<!DOCTYPE html>
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
 
<style>
* { font-family: Helvetica, Verdana, Arial; }
pre { white-space: pre-wrap; 
overflow-x: auto;
background-color: lightyellow;
font-family: Consolas, "Andale Mono WT", "Andale Mono", "Lucida Console", "Lucida Sans Typewriter", "DejaVu Sans Mono", "Bitstream Vera Sans Mono", "Liberation Mono", "Nimbus Mono L", Monaco, "Courier New", Courier, monospace; }
h1{ color: #b44;}
.accordion {
  background-color: #eee;
  color: #444;
  cursor: pointer;
  padding: 18px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
  transition: 0.4s;
}

.active, .accordion:hover {
  background-color: #ccc;
}

.accordion:after {
  content: '\002B';
  color: #777;
  font-weight: bold;
  float: right;
  margin-left: 5px;
}

.active:after {
  content: "\2212";
}

.panel {
  padding: 0 18px;
  background-color: white;
  max-height: 0;
  overflow: hidden;
  transition: max-height 0.2s ease-out;
}

/* Table */

table {
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

td, th {
  border: 1px solid #eee;
  text-align: left;
  padding: 8px;
}
th {
  background-color: #ddd;
}
tr:nth-child(odd) {
  background-color: #eee;
}

.bar .bar-item{padding:8px 16px;float:left;width:auto;border:none;display:block;outline:0}
.bar .button{white-space:normal}
.bar .dropdown-hover,.bar .dropdown-click{position:static;float:left}
.bar-block .bar-item{width:100%;display:block;padding:8px 16px;text-align:left;border:none;white-space:normal;float:none;outline:0}
.bar-block .dropdown-hover .button,.bar-block .dropdown-click .button{width:100%;text-align:left;padding:8px 16px}
.bar-block .dropdown-hover .dropdown-content,.bar-block .dropdown-click .dropdown-content{min-width:100%}
.bar-block .dropdown-hover,.bar-block .dropdown-click{width:100%}
.bar-block.center .bar-item{text-align:center}.block{display:block;width:100%}
.bar-block.center .bar-item{text-align:center}.block{display:block;width:100%}
.bar{width:100%;overflow:hidden}.center .bar{display:inline-block;width:auto}
.bar{width:100%;overflow:hidden}.center .bar{display:inline-block;width:auto}
.black,.hover-black:hover{color:#fff!important;background-color:#000!important}
.btn,.button{-webkit-touch-callout:none;-webkit-user-select:none;-khtml-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}   
.btn,.button{border:none;display:inline-block;padding:8px 16px;vertical-align:middle;overflow:hidden;text-decoration:none;color:inherit;background-color:inherit;text-align:center;cursor:pointer;white-space:nowrap}
.button:hover{color:#000!important;background-color:#ccc!important}
.cell-row:before,.cell-row:after,.clear:after,.clear:before,.bar:before,.bar:after{content:"";display:table;clear:both}
.cell-row:before,.cell-row:after,.clear:after,.clear:before,.bar:before,.bar:after{content:"";display:table;clear:both}
.cell-row:before,.cell-row:after,.clear:after,.clear:before,.bar:before,.bar:after{content:"";display:table;clear:both} 
.container:after,.container:before,.panel:after,.panel:before,.row:after,.row:before,.row-padding:after,.row-padding:before,
.disabled,.btn:disabled,.button:disabled{cursor:not-allowed;opacity:0.3}.disabled *,:disabled *{pointer-events:none}
.display-topleft{position:absolute;left:0;top:0}.display-topright{position:absolute;right:0;top:0}
.dropdown-hover.mobile,.dropdown-hover.mobile .btn,.dropdown-hover.mobile .button,.dropdown-click.mobile,.dropdown-click.mobile .btn,.dropdown-click.mobile .button{width:100%}}
.dropdown-hover:hover > .button:first-child,.dropdown-click:hover > .button:first-child{background-color:#ccc;color:#000}
.hide-small{display:none!important}.mobile{display:block;width:100%!important}.bar-item.mobile,.dropdown-hover.mobile,.dropdown-click.mobile{text-align:center}
.tooltip,.display-container{position:relative}.tooltip .text{display:none}.tooltip:hover .text{display:inline-block}

.responsive {
  width: 100%;
  height: auto;
}
</style>
</head>
<body>
<h1>Dadaist2 execution log</h1>
<p>
This is not a report with the results of the pipeline, but the log of the events occurred to generate it.
</p>
<button class="accordion">Summary</button>
<div class="panel">

<!--summary -->
 

<div class="bar black">
  <button class="bar-item button" onclick="openCity('Summary')">Summary</button>
  <button class="bar-item button" onclick="openCity('Parameters')">Parameters</button>
  <button class="bar-item button" onclick="openCity('Host')">Machine</button>
</div>

<div id="Summary" class="container display-container city">
  <span onclick="this.parentElement.style.display='none'"
  class="button large display-topright">&times;</span>

  <h2>Summary</h2>
  <p>This reports contain technical details on the execution of 
  <a href="https://quadram-institute-bioscience.github.io/dadaist2">Dadaist2</a>.</p>

<table>
  <tr>
    <th>Parameter</th>
    <th>Value</th>
  </tr>
  <tr>
    <td>Input</td>
    <td>/Users/telatina/git/dadaist2/data/16S</td>
  </tr>
  <tr>
    <td>Output</td>
    <td>/Users/telatina/git/dadaist2/output/test-log</td>
  </tr>
  <tr>
    <td>Timestamp</td>
    <td>20210304 14:17:36</td>
  </tr>
  <tr>
    <td>Execution time</td>
    <td>1m 10s</td>
  </tr>

</table>


</div>

<div id="Parameters" class="container display-container city" style="display:none">
  <span onclick="this.parentElement.style.display='none'"
  class="button large display-topright">&times;</span>
  <h2>Parameters</h2>  
	<table>
  <tr>
    <th>Parameter</th>
    <th>Value</th>
  </tr>
	<br><tr>
    	<td>Database</td>
    	<td>/Users/telatina/git/dadaist2/skip</td>
  	</tr><br><tr>
    	<td>Metadata</td>
    	<td>/Users/telatina/git/dadaist2</td>
  	</tr><br><tr>
    	<td>Threads</td>
    	<td>/Users/telatina/git/dadaist2/8</td>
  	</tr>
	</table>
</div>

<div id="Host" class="container display-container city" style="display:none">
  <span onclick="this.parentElement.style.display='none'"
  class="button large display-topright">&times;</span>
  <h2>Machine details</h2>
<table>
  <tr>
    <th>Timestamp</th>
    <th>Event</th>
  </tr>
  <tr>
    <td>Processors</td>
    <td>8</td>
  </tr>
  <tr>
    <td>RAM</td>
    <td>16 Gb</td>
  </tr>
  <tr>
    <td>OS</td>
    <td><table><tr><td>Model Name</td><td>MacBook Pro</td></tr>
<tr><td>Model Identifier</td><td>MacBookPro16,1</td></tr>
<tr><td>Processor Name</td><td>8-Core Intel Core i9</td></tr>
<tr><td>Processor Speed</td><td>2.3 GHz</td></tr>
<tr><td>Number of Processors</td><td>1</td></tr>
<tr><td>Total Number of Cores</td><td>8</td></tr>
<tr><td>L2 Cache (per Core)</td><td>256 KB</td></tr>
<tr><td>L3 Cache</td><td>16 MB</td></tr>
<tr><td>Hyper-Threading Technology</td><td>Enabled</td></tr>
<tr><td>Memory</td><td>16 GB</td></tr>
<tr><td>System Firmware Version</td><td>1554.60.15.0.0 (iBridge</td></tr>
<tr><td>Serial Number (system)</td><td>C02ZL5N9MD6N</td></tr>
<tr><td>Hardware UUID</td><td>8E0231F9-97C9-5983-93C9-B5825D7A0E67</td></tr>
<tr><td>Provisioning UDID</td><td>8E0231F9-97C9-5983-93C9-B5825D7A0E67</td></tr>
<tr><td>Activation Lock Status</td><td>Enabled</td></tr>
</table>
</td>
  </tr>
</table>
</div>

 <br>
</div>
<!--/summary-->





<button class="accordion">Read trimming (process)</button>
<div class="panel">
  <br><p><h3>Shell command <span style="color: #b44">fastp</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:47</em> for <em>0.59s</em>.</p><p><strong>Command</strong><pre>fastp -w 8 -i "/Users/telatina/git/dadaist2/data/16S/A01_S0_L001_R1_001.fastq.gz" -I "/Users/telatina/git/dadaist2/data/16S/A01_S0_L001_R2_001.fastq.gz" -o "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl"/for/A01_R1.fastq.gz -O  "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl"/rev/A01_R2.fastq.gz   --trim_front1 20 --trim_front2  20  --detect_adapter_for_pe --disable_quality_filtering --n_base_limit  1   --json "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/A01.json" --html /dev/null  --trim_tail1 0 --trim_tail2 0 --length_required 110 </pre></p>
<p><strong>Standard output</strong><pre><p><strong>Standard error</strong><pre>Detecting adapter sequence for read1...
No adapter detected for read1

Detecting adapter sequence for read2...
No adapter detected for read2

Read1 before filtering:
total reads: 6137
total bases: 1847237
Q20 bases: 1648961(89.2663%)
Q30 bases: 1490142(80.6687%)

Read2 before filtering:
total reads: 6137
total bases: 1847237
Q20 bases: 1577946(85.422%)
Q30 bases: 1382606(74.8472%)

Read1 after filtering:
total reads: 6105
total bases: 1689825
Q20 bases: 1504522(89.0342%)
Q30 bases: 1353780(80.1136%)

Read2 aftering filtering:
total reads: 6105
total bases: 1689825
Q20 bases: 1439363(85.1782%)
Q30 bases: 1251309(74.0496%)

Filtering result:
reads passed filter: 12210
reads failed due to low quality: 0
reads failed due to too many N: 0
reads failed due to too short: 64
reads with adapter trimmed: 1090
bases trimmed due to adapters: 64672

Duplication rate: 7.27273%

Insert size peak (evaluated by paired-end reads): 465

JSON report: /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/A01.json
HTML report: /dev/null

fastp -w 8 -i /Users/telatina/git/dadaist2/data/16S/A01_S0_L001_R1_001.fastq.gz -I /Users/telatina/git/dadaist2/data/16S/A01_S0_L001_R2_001.fastq.gz -o /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/for/A01_R1.fastq.gz -O /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/rev/A01_R2.fastq.gz --trim_front1 20 --trim_front2 20 --detect_adapter_for_pe --disable_quality_filtering --n_base_limit 1 --json /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/A01.json --html /dev/null --trim_tail1 0 --trim_tail2 0 --length_required 110 
fastp v0.20.1, time used: 1 seconds</p>
<br><p><h3>Shell command <span style="color: #b44">fastp</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:48</em> for <em>0.54s</em>.</p><p><strong>Command</strong><pre>fastp -w 8 -i "/Users/telatina/git/dadaist2/data/16S/A02_S0_L001_R1_001.fastq.gz" -I "/Users/telatina/git/dadaist2/data/16S/A02_S0_L001_R2_001.fastq.gz" -o "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl"/for/A02_R1.fastq.gz -O  "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl"/rev/A02_R2.fastq.gz   --trim_front1 20 --trim_front2  20  --detect_adapter_for_pe --disable_quality_filtering --n_base_limit  1   --json "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/A02.json" --html /dev/null  --trim_tail1 0 --trim_tail2 0 --length_required 110 </pre></p>
<p><strong>Standard output</strong><pre><p><strong>Standard error</strong><pre>Detecting adapter sequence for read1...
No adapter detected for read1

Detecting adapter sequence for read2...
No adapter detected for read2

Read1 before filtering:
total reads: 5414
total bases: 1629614
Q20 bases: 1454370(89.2463%)
Q30 bases: 1311878(80.5024%)

Read2 before filtering:
total reads: 5414
total bases: 1629614
Q20 bases: 1412945(86.7043%)
Q30 bases: 1240421(76.1175%)

Read1 after filtering:
total reads: 5399
total bases: 1508588
Q20 bases: 1338035(88.6945%)
Q30 bases: 1200838(79.6001%)

Read2 aftering filtering:
total reads: 5399
total bases: 1508588
Q20 bases: 1299416(86.1346%)
Q30 bases: 1130799(74.9574%)

Filtering result:
reads passed filter: 10798
reads failed due to low quality: 0
reads failed due to too many N: 0
reads failed due to too short: 30
reads with adapter trimmed: 648
bases trimmed due to adapters: 23266

Duplication rate: 14.6341%

Insert size peak (evaluated by paired-end reads): 0

JSON report: /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/A02.json
HTML report: /dev/null

fastp -w 8 -i /Users/telatina/git/dadaist2/data/16S/A02_S0_L001_R1_001.fastq.gz -I /Users/telatina/git/dadaist2/data/16S/A02_S0_L001_R2_001.fastq.gz -o /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/for/A02_R1.fastq.gz -O /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/rev/A02_R2.fastq.gz --trim_front1 20 --trim_front2 20 --detect_adapter_for_pe --disable_quality_filtering --n_base_limit 1 --json /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/A02.json --html /dev/null --trim_tail1 0 --trim_tail2 0 --length_required 110 
fastp v0.20.1, time used: 0 seconds</p>
<br><p><h3>Shell command <span style="color: #b44">fastp</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:49</em> for <em>0.57s</em>.</p><p><strong>Command</strong><pre>fastp -w 8 -i "/Users/telatina/git/dadaist2/data/16S/F99_S0_L001_R1_001.fastq.gz" -I "/Users/telatina/git/dadaist2/data/16S/F99_S0_L001_R2_001.fastq.gz" -o "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl"/for/F99_R1.fastq.gz -O  "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl"/rev/F99_R2.fastq.gz   --trim_front1 20 --trim_front2  20  --detect_adapter_for_pe --disable_quality_filtering --n_base_limit  1   --json "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/F99.json" --html /dev/null  --trim_tail1 0 --trim_tail2 0 --length_required 110 </pre></p>
<p><strong>Standard output</strong><pre><p><strong>Standard error</strong><pre>Detecting adapter sequence for read1...
No adapter detected for read1

Detecting adapter sequence for read2...
No adapter detected for read2

Read1 before filtering:
total reads: 4553
total bases: 1370453
Q20 bases: 1233254(89.9888%)
Q30 bases: 1118036(81.5815%)

Read2 before filtering:
total reads: 4553
total bases: 1370453
Q20 bases: 1175086(85.7443%)
Q30 bases: 1025042(74.7959%)

Read1 after filtering:
total reads: 4536
total bases: 1267283
Q20 bases: 1134520(89.5238%)
Q30 bases: 1023409(80.7562%)

Read2 aftering filtering:
total reads: 4536
total bases: 1267283
Q20 bases: 1078894(85.1344%)
Q30 bases: 932192(73.5583%)

Filtering result:
reads passed filter: 9072
reads failed due to low quality: 0
reads failed due to too many N: 0
reads failed due to too short: 34
reads with adapter trimmed: 432
bases trimmed due to adapters: 21554

Duplication rate: 10.8911%

Insert size peak (evaluated by paired-end reads): 0

JSON report: /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/F99.json
HTML report: /dev/null

fastp -w 8 -i /Users/telatina/git/dadaist2/data/16S/F99_S0_L001_R1_001.fastq.gz -I /Users/telatina/git/dadaist2/data/16S/F99_S0_L001_R2_001.fastq.gz -o /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/for/F99_R1.fastq.gz -O /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/rev/F99_R2.fastq.gz --trim_front1 20 --trim_front2 20 --detect_adapter_for_pe --disable_quality_filtering --n_base_limit 1 --json /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/F99.json --html /dev/null --trim_tail1 0 --trim_tail2 0 --length_required 110 
fastp v0.20.1, time used: 0 seconds</p>

</div>

<button class="accordion">DADA2 Denoising (process)</button>
<div class="panel">
  <p><h3>Shell command <span style="color: #b44">Rscript</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:49</em> for <em>17.84s</em>.</p><p><strong>Command</strong><pre>Rscript --vanilla /Users/telatina/git/dadaist2/bin/D2-dada.R /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/for /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/rev /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/dada2/dada2.tsv /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/dada2/stats.tsv /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/for/filtered /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/rev/filtered 265 207 0 0 1 1.5 10 consensus 1 8 10000 /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl do_plots skip no 0</pre></p>
<p><strong>Standard output</strong><pre>R version 4.0.3 (2020-10-10) 
#DADA2: 1.18.0 / Rcpp: 1.0.6 / RcppParallel: 5.0.3 
2021-03-04 14:17:56 	[1] Filtering reads ...
2021-03-04 14:18:04 	[2] Learning Error Rates
1579400 total bases in 5960 reads from 3 samples will be used for learning the error rates.
1233720 total bases in 5960 reads from 3 samples will be used for learning the error rates.
2021-03-04 14:18:06 	[3] Denoise remaining samples ...
2021-03-04 14:18:07	[4] Remove chimeras (method = consensus)
2021-03-04 14:18:07 	[5.1] Taxonomy (SKIPPED)
2021-03-04 14:18:07 	[6] Write output
	 *  /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl/dada2/dada2.tsv <p><strong>Standard error</strong><pre>Loading required package: Rcpp</p>

</div>

<button class="accordion">Taxonomy (process)</button>
<div class="panel">
  <p><h3>Shell command <span style="color: #b44">/Users/telatina/git/dadaist2/bin/dadaist2-assigntax</span> returned 0</h3><p>Executed at <em>2021-03-04 14:18:07</em> for <em>37.54s</em>.</p><p><strong>Command</strong><pre>/Users/telatina/git/dadaist2/bin/dadaist2-assigntax -i "/Users/telatina/git/dadaist2/output/test-log/rep-seqs.fasta" -o "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl" -r "refs/SILVA_SSU_r138_2019.RData" -t "8"</pre></p>
<p><strong>Standard output</strong><pre><p><strong>Standard error</strong><pre>[35;1m DADAIST2 Assign Taxonomy[0m

 Input file:  /Users/telatina/git/dadaist2/output/test-log/rep-seqs.fasta
 Reference:   refs/SILVA_SSU_r138_2019.RData
 Output dir:  /var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl
 Threads:     8

 * Using DECIPHER
[2021-03-04 14:18:07] Shell> Rscript --vanilla "/Users/telatina/git/dadaist2/bin/D2-decipher.R" "/Users/telatina/git/dadaist2/output/test-log/rep-seqs.fasta" "refs/SILVA_SSU_r138_2019.RData" "/var/folders/qf/64kyljyn0njdfbz8nyrgjv8hxrrkbt/T/dadaist2_eaYrbl" "8"
[2021-03-04 14:18:45]     +> Elapsed time: 37.37s; Exit status: 0;</p>

</div>


<button class="accordion">Initial checks</button>
<div class="panel">
	<br><p><h3>Shell command <span style="color: #b44">echo</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:37</em> for <em>2.60s</em>.</p><p><strong>Command</strong><pre>echo "library(DECIPHER)" | R --no-save > /dev/null</pre></p>
<p><strong>Standard output</strong><pre><p><strong>Standard error</strong><pre>Loading required package: Biostrings
Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: â€˜BiocGenericsâ€™

The following objects are masked from â€˜package:parallelâ€™:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from â€˜package:statsâ€™:

    IQR, mad, sd, var, xtabs

The following objects are masked from â€˜package:baseâ€™:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors
Loading required package: stats4

Attaching package: â€˜S4Vectorsâ€™

The following object is masked from â€˜package:baseâ€™:

    expand.grid

Loading required package: IRanges
Loading required package: XVector

Attaching package: â€˜Biostringsâ€™

The following object is masked from â€˜package:baseâ€™:

    strsplit

Loading required package: RSQLite</p>
<br><p><h3>Shell command <span style="color: #b44">/Users/telatina/git/dadaist2/bin/dadaist2-assigntax</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:40</em> for <em>0.16s</em>.</p><p><strong>Command</strong><pre>/Users/telatina/git/dadaist2/bin/dadaist2-assigntax --version</pre></p>
<p><strong>Standard output</strong><pre>dadaist2-assigntax 1.0.5<p><strong>Standard error</strong><pre>[35;1m DADAIST2 Assign Taxonomy[0m</p>
<br><p><h3>Shell command <span style="color: #b44">Rscript</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:40</em> for <em>0.01s</em>.</p><p><strong>Command</strong><pre>Rscript  --version 2>&1</pre></p>
<p><strong>Standard output</strong><pre>R scripting front-end version 4.0.3 (2020-10-10)<p><strong>Standard error</strong><pre></p>
<br><p><h3>Shell command <span style="color: #b44">clustalo</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:40</em> for <em>0.01s</em>.</p><p><strong>Command</strong><pre>clustalo --version 2>&1</pre></p>
<p><strong>Standard output</strong><pre>1.2.4<p><strong>Standard error</strong><pre></p>
<br><p><h3>Shell command <span style="color: #b44">echo</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:40</em> for <em>6.89s</em>.</p><p><strong>Command</strong><pre>echo "library(dada2)" | R --no-save > /dev/null</pre></p>
<p><strong>Standard output</strong><pre><p><strong>Standard error</strong><pre>Loading required package: Rcpp</p>
<br><p><h3>Shell command <span style="color: #b44">/Users/telatina/git/dadaist2/bin/dadaist2-exporter</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:47</em> for <em>0.08s</em>.</p><p><strong>Command</strong><pre>/Users/telatina/git/dadaist2/bin/dadaist2-exporter --version</pre></p>
<p><strong>Standard output</strong><pre>dadaist2-exporter 1.0.0<p><strong>Standard error</strong><pre></p>
<br><p><h3>Shell command <span style="color: #b44">fastp</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:47</em> for <em>0.01s</em>.</p><p><strong>Command</strong><pre>fastp --version 2>&1</pre></p>
<p><strong>Standard output</strong><pre>fastp 0.20.1<p><strong>Standard error</strong><pre></p>
<br><p><h3>Shell command <span style="color: #b44">fasttree</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:47</em> for <em>0.02s</em>.</p><p><strong>Command</strong><pre>fasttree 2>&1 | head -n 1 | cut -c 11-</pre></p>
<p><strong>Standard output</strong><pre>FastTree version 2.1.10 Double precision (No SSE3):<p><strong>Standard error</strong><pre></p>
<br><p><h3>Shell command <span style="color: #b44">fu-primers</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:47</em> for <em>0.02s</em>.</p><p><strong>Command</strong><pre>fu-primers --version 2>&1</pre></p>
<p><strong>Standard output</strong><pre>fu-primers 1.0<p><strong>Standard error</strong><pre></p>
<br><p><h3>Shell command <span style="color: #b44">/Users/telatina/git/dadaist2/bin/makeSampleSheet</span> returned 0</h3><p>Executed at <em>2021-03-04 14:17:47</em> for <em>0.08s</em>.</p><p><strong>Command</strong><pre>/Users/telatina/git/dadaist2/bin/makeSampleSheet --version</pre></p>
<p><strong>Standard output</strong><pre>makeSampleSheet 1.0.0<p><strong>Standard error</strong><pre></p>

</div>


<script>
var acc = document.getElementsByClassName("accordion");
var i;

for (i = 0; i < acc.length; i++) {
  acc[i].addEventListener("click", function() {
    this.classList.toggle("active");
    var panel = this.nextElementSibling;
    if (panel.style.maxHeight) {
      panel.style.maxHeight = null;
    } else {
      panel.style.maxHeight = panel.scrollHeight + "px";
    } 
  });
}
</script>


<script>
function openCity(cityName) {
  var i;
  var x = document.getElementsByClassName("city");
  for (i = 0; i < x.length; i++) {
    x[i].style.display = "none";  
  }
  document.getElementById(cityName).style.display = "block";  
}
</script>
</body>



</html>

